code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -241,5 +232,26 @@
 /obj/item/external_lens/shield
 	name = "external lens: barricade projector"
 	desc = "Projects holobarricades which temporary absorb projectiles, watch out as even your target might use them as cover."
-	stored_ammo_type = /obj/item/ammo_casing/energy/laser/lowenergy
-	overlay = "shield"
\ No newline at end of file
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/shield
+	overlay = "shield"
+
+/datum/component/extralasers //will move it to another file when it works
+	var/obj/item/external_lens/lens
+/datum/component/extralasers/Initialize()

```

Comment:
Please put some whitespace before procs/types to make things a little more readable. It's also how we do it most everywhere anyway.

Reply:
Haven't I told you this before